### ISSUE TITLE
Null checking when find pin by UID fails

### ISSUE THREAD

**Issue by Fattorino** [Fattorino/ImNodeFlow#7]
Getting a Pin that doesn't exist simply crashes the program

**Comment by avlec** [comment]
Wouldn't this be the expected behaviour? Un-handled exception leads to program crash?

One improvement that can also be made in this area is making the `getInVal` match up with `addOUT`/`addIN` via taking a std::string.

**Comment by Fattorino** [comment]
Yes, this is the expected behavior but I'm not happy with how I implemented Pin's value retrieval so I'll do a little experimentation to see if I find something better.

Also, `getInVal` is a template so I'm not sure what you mean by that.

**Comment by Fattorino** [comment]
commit 0cbbf51 closes this issue

**Comment by avlec** [comment]
Might have been mixing that up with something else I'll take another gander later

**Comment by Fattorino** [comment]
`addIN` and `addOUT` hash a `std::string` to create a UID, but when getting a value from a template something like `"foo"` becomes a `char*` and therefore the hash doesn't match even if the string does. So I created that one specialization to go back to `std::string`.
Regarding the null checking, I decided to simply add an assertion to make it crash more nicely. Because I agree that a crash is the intended behavior. I'll also add more ways to check if a pin exists because they might be useful.
